apps/src/templates/studioHomepages/TeacherHomepage.jsx

```diff
@@ -286,7 +290,10 @@ export const UnconnectedTeacherHomepage = ({
           />
         )}
         <TeacherResources />
-        {showIncubatorBanner && <IncubatorBanner />}
+        {showIncubatorBanner &&
+          currentGlobalConfigurationFor('teacherHomepage.incubatorBanner') && (
```

Comment:
What if, instead of wrapping the component in configuration flags in the parent component, we configure the components themselves based on the current page path?

### Global Region Configs
```javascript
const globalRegionConfigs = {
  header: {},
  footer: {},
  pages: {
    '.*/home.*': { // page path mask
      incubatorBanner: false, // componentId config
    },
  },
};
```

### Global Region Wrapper
```javascript
const globalRegionWrapper = (WrappedComponent, componentId) => {
  // Retrieves the current page Global Region config based on the page path mask.
  const pageConfigKey = Object.keys(globalRegionConfigs.pages).find(
    pageMask => RexExp(pageMask).test(window.location.pathname)
  );
  const pageConfig = globalRegionConfigs.pages[pageConfigKey];

  // Renders the component if it is enabled for the page
  return !pageConfig || pageConfig[componentId]
    ? props => WrappedComponent({...props})
    : null;
};
```

### `apps/src/templates/studioHomepages/IncubatorBanner.jsx`
```javascript
const IncubatorBanner = () => {
  ...
};
export default globalRegionWrapper(IncubatorBanner, 'incubatorBanner');
```